sql/core/src/test/scala/org/apache/spark/sql/execution/SortSuite.scala

```diff
@@ -119,4 +120,20 @@ class SortSuite extends SparkPlanTest with SharedSQLContext {
       )
     }
   }
+
+  test("SPARK-22716: SortExec should not use a global variable to reference to itself") {
+    val ctx = new CodegenContext
+    val df = Seq(("a", 1)).toDF("a", "b").sort("a")
+    val plan = df.queryExecution.executedPlan
+    val sortExec = plan.children.head.asInstanceOf[SortExec]
+    sortExec.produce(ctx, plan.asInstanceOf[CodegenSupport])
+    // we expect 8 global variables:
```

Comment:
This test and that one in `BroadcastJoinSuite` seems a bit overkill.

Reply:
Thanks, I was also not very happy with them too, but I have not found a better way to test them. Do you have any idea/suggestion? Thanks.

Reply:
If we make sure `addReferenceMinorObj` work well, the two tests may not be necessary, IMHO. Let's wait others options.

Reply:
+1, we don't need new tests if we trust `addReferenceMinorObj`

Reply:
thanks, I'll remove them